nautobot/ipam/views.py

```diff
@@ -1106,73 +1107,81 @@ def list(self, request, *args, **kwargs):
 #
 
 
-class VLANGroupListView(generic.ObjectListView):
+class VLANGroupUIViewSet(NautobotUIViewSet):
+    bulk_update_form_class = forms.VLANGroupBulkEditForm
+    filterset_class = filters.VLANGroupFilterSet
+    filterset_form_class = forms.VLANGroupFilterForm
+    form_class = forms.VLANGroupForm
     queryset = VLANGroup.objects.all()
-    filterset = filters.VLANGroupFilterSet
-    filterset_form = forms.VLANGroupFilterForm
-    table = tables.VLANGroupTable
+    serializer_class = serializers.VLANGroupSerializer
+    table_class = tables.VLANGroupTable
 
-
-class VLANGroupView(generic.ObjectView):
-    queryset = VLANGroup.objects.all()
-
-    def get_extra_context(self, request, instance):
-        vlans = (
-            VLAN.objects.restrict(request.user, "view")
-            .filter(vlan_group=instance)
-            .prefetch_related(Prefetch("prefixes", queryset=Prefix.objects.restrict(request.user)))
-        )
-        vlans_count = vlans.count()
-
-        data_transform_callback = get_add_available_vlans_callback(show_available=True, vlan_group=instance)
-
-        vlan_table = tables.VLANDetailTable(
-            vlans, exclude=["vlan_group"], data_transform_callback=data_transform_callback
+    object_detail_content = object_detail.ObjectDetailContent(
+        panels=(
+            object_detail.ObjectFieldsPanel(
+                weight=100,
+                section=SectionChoices.LEFT_HALF,
+                fields="__all__",
+            ),
+            object_detail.ObjectsTablePanel(
+                weight=100,
+                section=SectionChoices.RIGHT_HALF,
+                context_table_key="vlan_table",
+                related_field_name="vlan_group",
+                enable_bulk_actions=True,
+                add_button_route=None,
+                form_id="vlan_form",
+                footer_buttons=[
+                    object_detail.FormButtonNoPK(
+                        link_name="ipam:vlan_bulk_edit",
+                        label="Edit Selected",
+                        color=ButtonActionColorChoices.EDIT,
+                        icon="mdi-pencil",
+                        size="xs",
+                        form_id="vlan_form",
+                        weight=200,
+                    ),
+                    object_detail.FormButtonNoPK(
+                        link_name="ipam:vlan_bulk_delete",
+                        label="Delete Selected",
+                        color=ButtonActionColorChoices.DELETE,
+                        icon="mdi-trash-can-outline",
+                        size="xs",
+                        form_id="vlan_form",
+                        weight=100,
+                    ),
+                ],
```

Comment:
Feels like there should be a pattern for automatically including the relevant bulk action buttons on an ObjectsTablePanel, without needing to explicitly describe them like this. Is this the first ObjectsTablePanel we've encountered that had a need for these buttons?

Reply:
I feel like this is the time I have seen it personally, as far as BulkEdit and BulkDelete buttons in a related object table in a detail view.